src/modding/ModManager.cs

```diff
@@ -784,18 +1170,40 @@ private void ApplyChanges()
             return;
         }
 
+        var checkResult = ModLoader.IsValidModList(enabledMods);
+        if (checkResult.ErrorType >= 0)
+        {
+            LoadEnabledMods();
+        }
+        else
+        {
+            var warningText = new LocalizedString("MOD_LOAD_ERROR_WARNING",
+                ModHelpers.CheckResultToString(checkResult, enabledMods));
+
+            loadWarningDialog.DialogText = warningText.ToString();
+            loadWarningDialog.PopupCenteredShrink();
+        }
+    }
+
+    private void LoadEnabledMods()
+    {
+        if (notEnabledMods == null || enabledMods == null)
+        {
+            return;
```

Comment:
Add GD.PrintErrs here about invalid state (and also all the other places that has these checks for notEnabledMods and enabledMods). Actually maybe it would be better to make notEnabledMods and enabledMods into readonly lists that *always* exist. I think that would remove a *ton* of error checking and the case where the list content is replaced entirely can be just handled with clear and AddRange.